projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php

```diff
@@ -907,6 +915,20 @@ function wpcom_launchpad_is_paid_newsletter_enabled() {
 	return wpcom_launchpad_has_goal_paid_subscribers() && apply_filters( 'wpcom_launchpad_intent_paid_newsletter_enabled', false );
 }
 
+/**
+ * Checks if site has newsletter or write as site intent.
+ *
+ * @return bool True if the task list is enabled, false otherwise.
+ */
+function wpcom_launchpad_is_newsletter_or_write_intent() {
```

Comment:
Putting aside the logic issue with the condition in the function, I think we may need to revisit the naming for this function. As far as I can tell, the function should return true when the intent is _not_ `write` or `newsletter`. Can we update the naming so the naming aligns with what we're checking.
```suggestion
function wpcom_launchpad_is_not_newsletter_or_write_intent() {
```

Reply:
@daledupreez The intent is to return true if intent is either write or newsletter. So I think I just need to flip the true/false returns.

```
function wpcom_launchpad_is_newsletter_or_write_intent() {
	$intent = get_option( 'site_intent', false );
	if ( 'newsletter' === $intent || 'write' === $intent ) {
		return true;
	}

	return false;
}
```

I'll update. One odd thing I need to explore is why this has been working as intended when tested alongside my frontend PR.

Reply:
I'm testing this via dev console know. No matter whether I update that condition or not, I get back the same expected checklist. Maybe I misunderstand the purpose of that callback. That or there's a bug where it's not applying as expected.

Reply:
Just a note that the method for which this comment was added has been removed because I've updated this PR to be generic, and to show the earn launchpad for all sites, not just newsletter / blog / write.